File: Java/1_Array/LC_1431_KidsWithTheGreatestNumberOfCandies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        int max = 0;
        for (int num : candies) {
            max = Math.max(max, num);
        }

        List<Boolean> res = new ArrayList<>();
        for (int num : candies) {
            res.add(num + extraCandies >= max);
        }

        return res;
    }
}

public class LC_1431_KidsWithTheGreatestNumberOfCandies {

}

File: Swift/Leetcode.playground/Sources/1_Array/LC_1431_KidsWithTheGreatestNumberOfCandies.swift
class Solution_1431 {
    func kidsWithCandies(_ candies: [Int], _ extraCandies: Int) -> [Bool] {
        guard let max = candies.max() else { return [] }
        return candies.map { return $0 + extraCandies >= max }
    }
}

public class LC_1431_KidsWithTheGreatestNumberOfCandies : Solution_1431 {
    public override init() {}
}
